includes/subscriptions/class-wc-payments-product-service.php

```diff
@@ -149,6 +150,17 @@ public static function has_wcpay_product_id( WC_Product $product ) : string {
 		return (bool) $product->get_meta( self::PRODUCT_ID_KEY, true );
 	}
 
+	/**
+	 * Prevents duplicate _wcpay_product_id when duplicating a subscription product.
+	 *
+	 * @param array $meta_keys The keys to exclude from the duplicate.
+	 * @return array Keys to exclude, including _wcpay_product_id.
+	 */
+	public static function exclude_meta_wcpay_product_id( $meta_keys ) {
+		$meta_keys[] = self::PRODUCT_ID_KEY;
```

Comment:
I noticed the following metadata is retained with duplicating subscription products:
- `self::PRODUCT_HASH_KEY` (`'_wcpay_product_hash'`)
- `self::PRICE_HASH_KEY` (`'_wcpay_product_price_hash'`)
- `self::PRICE_ID_KEY` (`'_wcpay_product_price_id'`)

Should we also exclude these?

Reply:
yeah, we should exclude all the product IDs and hashes. 

In #3088 we're introducing keys for `_live` and `_test` so I'm not sure how we want to handle that. Merge this PR, rebase @mattallan's PR and then update them there? Or merge Matt's, rebase this branch and then update them here.